build/bin/build-client-source

```diff
@@ -20,11 +20,45 @@ prepare_tarball() {
     if [ -f "${tarball}" ]; then
         echo ${tarball} already exists, skipping build
     else
-        git -C "${git_dir}" archive \
-            --format=tar.gz \
-            --prefix=${repo_name}-${commit}/ \
-            -o ${tarball} \
-            "${commit}"
+        if [ ${repo_name} = 'pmm-admin' -o ${repo_name} = 'pmm-agent' -o ${repo_name} = 'pmm-submodules' ]; then
+            if [ ${repo_name} = 'pmm-submodules' ]; then
+                repo_path="https://github.com/Percona-Lab/pmm-submodules.git"
+                commit="PMM-2.0"
+            else
+                repo_path=$(grep "${repo_name}.git" .gitmodules | awk '{print $3}')
+            fi
+            pushd ${source_dir}
+            git clone ${repo_path}
+            pushd ${repo_name}
+            git reset --hard
+            git clean -xdf
+            git checkout ${commit}
+            popd
+            rm -rf ${repo_name}-${commit}
+            mv ${repo_name} ${repo_name}-${commit}
+            if [ ${repo_name} = 'pmm-submodules' ]; then
+                pushd ${repo_name}-${commit}
+                for file in $(ls); do
+                    if [ $file != 'build' ]; then
+                        rm -rf $file
+                    fi
+                done
+                mv build/config ./
+                mv build/rpm ./
+                mv build/debian ./
+                rm -rf build
+                popd
+            fi
+            tar -zcvf ${repo_name}-${commit}.tar.gz ${repo_name}-${commit}
+            rm -rf ${repo_name}-${commit}
+            popd
+        else
+            git -C "${git_dir}" archive \
+                --format=tar.gz \
+                --prefix=${repo_name}-${commit}/ \
+                -o ${tarball} \
+                "${commit}"
+        fi
```

Comment:
My "bad smell" detector is on full alarm. What is going on there? Why we do another `git clone` if we already have everything cloned by submodules? Why we descent into submodules directories and remove stuff there? Really looks like an alien part there.

And Evgeniy – please add comments. The future generations (including you after two weeks of MySQL 8) will thank you.

Reply:
fixed.
But please note that once we don't do `git clone` and just work with data after submodule update we have the following issue during `make release` 
+ make release
fatal: Not a git repository (or any of the parent directories): .git
fatal: Not a git repository (or any of the parent directories): .git
env CGO_ENABLED=0 go build -v -o bin/pmm-agent -ldflags " \